mikrotik/resource_interface_wireguard_peer.go

```diff
@@ -0,0 +1,265 @@
+package mikrotik
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ddelnano/terraform-provider-mikrotik/client"
+	"github.com/hashicorp/terraform-plugin-framework-validators/int64validator"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/booldefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/int64default"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringdefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	tftypes "github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+type interfaceWireguardPeer struct {
+	client *client.Mikrotik
+}
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource                = &interfaceWireguardPeer{}
+	_ resource.ResourceWithConfigure   = &interfaceWireguardPeer{}
+	_ resource.ResourceWithImportState = &interfaceWireguardPeer{}
+)
+
+// NewInterfaceWireguardPeerResource is a helper function to simplify the provider implementation.
+func NewInterfaceWireguardPeerResource() resource.Resource {
+	return &interfaceWireguardPeer{}
+
+}
+
+func (i *interfaceWireguardPeer) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	i.client = req.ProviderData.(*client.Mikrotik)
+}
+
+// Metadata returns the resource type name.
+func (i *interfaceWireguardPeer) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_interface_wireguard_peer"
+}
+
+// Schema defines the schema for the resource.
```

Comment:
@DawoudMahmud one final comment. @jdelnano and I discussed that many of the computes attributes shouldn't be. He looked into it quite a bit and there seems to be an issue with the `tfplugindocs` doc generation.

I'd like to leave the following TODO comment here and then I think we are ready to merge this.

```suggestion
// Schema defines the schema for the resource.
// TODO: Reevaluate the Computed schema attributes and determine if that is correct
```

Below is a screenshot for how this issue manifests. @maksym-nazarenko I'd be curious to know if you've ever run into this issue.

![Screen Shot 2023-08-01 at 8 57 07 AM](https://github.com/ddelnano/terraform-provider-mikrotik/assets/5855593/9f30e623-7ba9-4ace-90bf-6eb6de2d51f5)

Reply:
Yup, definitely ran into this issue a few times which led to playing around with the computed field.